httprl support (for resources); mtm support for organizations, resources and datasets; minor changes

ziodave opened the pull request:
Hello,

Ok, so I finally spent time on the resources import and tested _httprl_. Results are quite promising, with ~31.000 resources (we're talking about the count):
- _w/ httprl_ less than 10 minutes
- w/o more than 2 hours

I left inside the code both the functions (the previous one has now prefix `_sync`) so that anybody can run a comparative test.

I also added support for mtm to Organizations, Resources and Datasets in order to support multiple importing threads.

This PR includes changes at https://github.com/NuCivic/dkan_migrate_base/pull/15 and https://github.com/NuCivic/dkan_migrate_base/pull/16.

Hope this helps, from now on I think I will make only minor changes. As usual I am available in case of need.

ziodave committed:
add support for organizations

ziodave committed:
add support for organizations in datasets

ziodave committed:
add support for HTTP Parallel requests

ziodave committed:
ignore IDEA project files

ziodave committed:
remove customizations from this files

ziodave committed:
remove httprl calls

ziodave committed:
remove httprl dependency

ziodave committed:
require the httprl library for parallel HTTP reqs

ziodave committed:
add support for httprl

ziodave committed:
add support for multi-threaded drush (using the *limit* parameter; the *page* parameter has been removed)

ziodave committed:
add support for multi-threaded drush (using the *limit* parameter; the *page* parameter has been removed)

ziodave committed:
move the list of fields to a separate function (like the other classes) to enable superclasses to override it

acouch commented:
mtm and httprl would be great to support. One concern is that Pantheon or other hosts that use containers and limit PHP processes won't work with httprl. Lets make sure that this doesn't break in those environments.

ziodave committed:
remove httprl dependency to maintain compatibility with environments that can't support it

ziodave committed:
when uninstalling, deregister the organization migration; use httprl only if available

ziodave committed:
fix comments

ziodave commented:
Ok, I removed the `httprl` dependency from `dkan_migrate_base.info` and make use of `httprl` only if available (otherwise uses the previous method) in `dkan_migrate_base.module`.

In `dkan_migrate_base.module` I also took the chance to add the deregistration of the *organization* migration, if it was previously registered.

Ideally in the `dkan_migrate_base_handle_response` function, we should handle connection failures (I added a TODO).

ziodave committed:
fix line removed setting extra value; only set extra key/value if value is set

ziodave commented:
I fix an issue in the code (I omitted the line setting the extra value). I also added a check that the value is not an empty string.
